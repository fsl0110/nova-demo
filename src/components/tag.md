### Select a HTML-Tag of your joice

#### With type strong

```jsx
<Tag type="strong">Text in strong tags content!</Tag>
```

#### With type h1

```jsx
<Tag type="h1">Text in h1 tags content!</Tag>
```

#### With type em

```jsx
<Tag type="em">Text in em tags content!</Tag>
```
