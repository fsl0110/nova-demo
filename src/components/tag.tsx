import React, { FC, ReactNode } from "react";
import { SharedProps } from "../typings";

export interface Props extends SharedProps {
  /** Nodes between this component **/
  children: ReactNode | ReactNode[];
  /** Type of a HTML-Tag this component will use as root tag**/
  type: keyof JSX.IntrinsicElements;
}

export const Tag: FC<Props> = ({ type, className, children, ...rest }) =>
  React.createElement(type, {
    dangerouslySetInnerHTML: { __html: children },
    className,
    ...rest
  });

export default Tag;
