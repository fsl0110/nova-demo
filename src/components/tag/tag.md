### Select a HTML-Tag of your joice

#### Tag type strong

```jsx
<Tag type="strong">Text in strong tag.</Tag>
```

#### Tag type h1

```jsx
<Tag type="h1">Text in h1 tag.</Tag>
```

#### Tag type em

```jsx
<Tag type="em">Text in em tag.</Tag>
```
