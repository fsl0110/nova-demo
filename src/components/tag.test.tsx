import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Tag, Props } from "./tag";

const testProps: Props = {
  children: "Some text",
  type: "h1"
};

const testCases: Array<[string, Props]> = [
  [
    "renders h1 tag",
    {
      type: "h1",
      children: "some text"
    }
  ],
  [
    "renders h2 tag",
    {
      type: "h1",
      children: "some text"
    }
  ],
  [
    "renders h3 tag",
    {
      type: "h1",
      children: "some text"
    }
  ]
];

describe("<Tag>", () => {
  it("renders without crashing", () => {
    shallow(<Tag {...testProps}>Some text</Tag>);
  });

  testCases.forEach(([item, options]) => {
    it(item, () => {
      const mock = shallow(<Tag {...options}>Some text</Tag>);
      expect(toJson(mock)).toMatchSnapshot();
    });
  });
});
