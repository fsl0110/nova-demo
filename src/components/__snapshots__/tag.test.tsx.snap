// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag> renders h1 tag 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Some text",
    }
  }
/>
`;

exports[`<Tag> renders h2 tag 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Some text",
    }
  }
/>
`;

exports[`<Tag> renders h3 tag 1`] = `
<h1
  dangerouslySetInnerHTML={
    Object {
      "__html": "Some text",
    }
  }
/>
`;
